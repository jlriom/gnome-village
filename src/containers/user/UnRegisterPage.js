import React from "react";
import { connect } from "react-redux";

import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";

const UnRegisterPage = ({ show, handleClose }) => {
	return (
		<Modal show={show} onHide={handleClose}>
			<Modal.Header closeButton>
				<Modal.Title>Unregister</Modal.Title>
			</Modal.Header>

			<Modal.Body>
				<p>Unregister</p>
			</Modal.Body>

			<Modal.Footer>
				<Button variant='secondary' onClick={handleClose}>
					Cancel
				</Button>
				<Button variant='primary'>Save</Button>
			</Modal.Footer>
		</Modal>
	);
};
export default connect(() => ({}))(UnRegisterPage);
