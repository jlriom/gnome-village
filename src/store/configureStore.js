import { createStore, applyMiddleware, compose } from "redux";
import reduxImmutableStateInvariant from "redux-immutable-state-invariant";
import thunk from "redux-thunk";
import logger from "redux-logger";

import rootReducer from "./reducers";

const middlewares = [thunk, logger, reduxImmutableStateInvariant()];

export default function configureStore(initialState) {
	const composeEnhancers =
		window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // add support for Redux dev tools

	return createStore(
		rootReducer,
		initialState,
		composeEnhancers(applyMiddleware(...middlewares))
	);
}
