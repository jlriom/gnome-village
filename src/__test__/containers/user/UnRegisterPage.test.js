import React from "react";
import UnRegisterPage from "../../../containers/user/UnRegisterPage";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "../../../store/configureStore";

const store = configureStore();

const props = {
    id: 0,
    name: "name",
    unRegisterme: jest.fn(),
    logout: jest.fn(),
    show: false,
    handleClose: jest.fn()
}

describe("<UnRegisterPage />", () => {
    it("renders ok", () => {
        const tree = renderer.create(
            <Provider store={store}>
                <UnRegisterPage {...props} />
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });
});

