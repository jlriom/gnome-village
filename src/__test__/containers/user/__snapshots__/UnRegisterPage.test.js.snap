// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnRegisterPage /> renders ok 1`] = `null`;
